narwhals/_pandas_like/expr.py

```diff
@@ -426,15 +432,27 @@ def func(df: PandasLikeDataFrame) -> list[PandasLikeSeries]:
                     )
                     raise ValueError(msg)
 
+                reverse = (
+                    self._kwargs.pop("reverse") if self._kwargs is not None else False
+                )
+                if reverse:
+                    msg = (
+                        "Cumulative operation with `reverse=True` is not supported in "
+                        "over context for pandas-like backend."
+                    )
+                    raise NotImplementedError(msg)
+
                 if self._function_name == "col->cum_count":
                     plx = self.__narwhals_namespace__()
                     df = df.with_columns(~plx.col(*self._root_names).is_null())
 
-                res_native = df._native_frame.groupby(list(keys), as_index=False)[
-                    self._root_names
-                ].transform(
-                    CUMULATIVE_FUNCTIONS_TO_PANDAS_EQUIVALENT[self._function_name]
-                )
+                res_native = getattr(
```

Comment:
Not a strong preference to use getattr, I realize now that also via transform it is easy to pass kwargs